libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -255,7 +255,7 @@ bool SyntaxChecker::visit(FunctionTypeName const& _node)
 
 	for (auto const& decl: _node.returnParameterTypeList()->parameters())
 		if (!decl->name().empty())
-			m_errorReporter.warning(decl->location(), "Naming function type return parameters is deprecated.");
+			m_errorReporter.syntaxError(decl->location(), "Return parameters in function types may not be named.");
```

Comment:
I'm fine with either text, though we kept the "is deprecated" error for most of the other features. I think it makes sense for the long standing features, but this probably wasn't used at all so it shouldn't make a difference.